ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -3065,18 +3082,149 @@ const triggerSet: TriggerSet<Data> = {
+      condition: (data) => data.seenSecondTethers === false,
+      infoText: (_data, _matches, output) => output.stackForTethers!(),
+      outputStrings: {
+        stackForTethers: {
+          en: 'Stack for Tethers',
+        },
+      },
+    },
+    {
+      id: 'P12S Ultima Ray 1',
+      type: 'StartsUsing',
+      netRegex: { id: '8330', source: 'Hemitheos' },
+      condition: (data) => data.phase === 'gaiaochos1',
+      infoText: (data, matches, output) => {
+        data.darknessClones.push(matches);
+        if (data.darknessClones.length !== 3)
+          return;
+
+        // during 'UAV' phase, the center of the circular arena is [100, 90]
+        const uavCenterX = 100;
+        const uavCenterY = 90;
+
+        const unsafeMap: Partial<Record<DirectionOutput8, DirectionOutput8>> = {
+          dirN: 'dirS',
+          dirNE: 'dirSW',
+          dirE: 'dirW',
+          dirSE: 'dirNW',
+          dirS: 'dirN',
+          dirSW: 'dirNE',
+          dirW: 'dirE',
+          dirNW: 'dirSE',
+        } as const;
+        let safeDirs = Object.keys(unsafeMap);
+        data.darknessClones.forEach((clone) => {
+          const x = parseFloat(clone.x);
+          const y = parseFloat(clone.y);
+          const cloneDir = Directions.xyTo8DirOutput(x, y, uavCenterX, uavCenterY);
+          const pairedDir = unsafeMap[cloneDir];
+          safeDirs = safeDirs.filter((dir) => dir !== cloneDir && dir !== pairedDir);
+        });
+        if (safeDirs.length !== 2)
+          return;
+        const [dir1, dir2] = safeDirs.sort();
+        if (dir1 === undefined || dir2 === undefined)
+          return;
+        return output.combined!({ dir1: output[dir1]!(), dir2: output[dir2]!() });
+      },
+      outputStrings: {
+        combined: {
+          en: '${dir1} / ${dir2} Safe',
+        },
+        ...Directions.outputStrings8Dir,
+      },
+    },
+    {
+      id: 'P12S Ultima Ray 2',
+      type: 'StartsUsing',
+      netRegex: { id: '8330', source: 'Hemitheos' },
+      condition: (data) => data.phase === 'gaiaochos2',
+      infoText: (_data, matches, output) => {
+        // during 'UAV' phase, the center of the circular arena is [100, 90]
+        const uavCenterX = 100;
+        const uavCenterY = 90;
+
+        const safeMap: Readonly<Partial<Record<DirectionOutput8, readonly DirectionOutput8[]>>> = {
+          // for each dir, identify the two dirs 90 degrees away
+          dirN: ['dirW', 'dirE'],
+          dirNE: ['dirNW', 'dirSE'],
+          dirE: ['dirN', 'dirS'],
+          dirSE: ['dirNE', 'dirSW'],
+          dirS: ['dirW', 'dirE'],
+          dirSW: ['dirNW', 'dirSE'],
+          dirW: ['dirN', 'dirS'],
+          dirNW: ['dirNE', 'dirSW'],
+        };
+
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+        const cloneDir = Directions.xyTo8DirOutput(x, y, uavCenterX, uavCenterY);
+        const [dir1, dir2] = safeMap[cloneDir] ?? [];
```

Comment:
```suggestion
        const safeMap: Record<DirectionOutput8, readonly DirectionOutput8[]> = {
          // for each dir, identify the two dirs 90 degrees away
          dirN: ['dirW', 'dirE'],
          dirNE: ['dirNW', 'dirSE'],
          dirE: ['dirN', 'dirS'],
          dirSE: ['dirNE', 'dirSW'],
          dirS: ['dirW', 'dirE'],
          dirSW: ['dirNW', 'dirSE'],
          dirW: ['dirN', 'dirS'],
          dirNW: ['dirNE', 'dirSW'],
          unknown: [],
        } as const;

        const x = parseFloat(matches.x);
        const y = parseFloat(matches.y);
        const cloneDir = Directions.xyTo8DirOutput(x, y, uavCenterX, uavCenterY);
        const [dir1, dir2] = safeMap[cloneDir];
```

Ok, then I'm gonna suggest this simpification though.